### PR TITLE
Implements Quandl support for Python

### PR DESCRIPTION
**Pull request by AlexCatarino**
Implements Quandl support for Python.
It was not possible to derive from Quandl in order to select the column. If the data did not have "close", it would thrown an exception since it would look for this word in a dictionary.
It is now possible to select the column.
See example QuandFuturesDataAlgorithm.py

### CODE REVIEW

File: Algorithm.Python/QuandlFuturesDataAlgorithm.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def OnData(self, data):
+        '''Data Event Handler: New data arrives here. "TradeBars" type is a dictionary of strings so you can access it by symbol.'''
+        if self.Portfolio.HoldStock: return
+     
+        self.SetHoldings(self.crude, 1);
+        self.Debug(str(self.Time) + str(" Purchased Crude Oil: ") + self.crude)
+
+
+class QuandlFuture(PythonQuandl):
```
Comment (jaredbroad) [inline Algorithm.Python/QuandlFuturesDataAlgorithm.py]: 👍 sweet!

File: Algorithm.Python/QCUWeatherBasedRebalancing.py [hunk 2]
```diff
@@ -83,8 +83,9 @@ def Reader(self, config, line, date, isLive):
         weather = Weather()
         weather.Symbol = config.Symbol
         weather.Time = datetime.strptime(data[0], '%Y-%m-%d') + timedelta(hours=20) # Make sure we only get this data AFTER trading day - don't want forward bias.
+        if not data[2]: return None
```
Comment (jaredbroad) [inline Algorithm.Python/QCUWeatherBasedRebalancing.py]: Please add comment here.

File: Algorithm.Python/QuandlFuturesDataAlgorithm.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    QuantConnect has a special deal with Quandl giving you access to Stevens Continuous Futurs (SCF) for free.
+    If you'd like to download SCF for local backtesting, you can download it through Quandl.com.'''
+
+    def Initialize(self):
+        ''' Initialize the data and resolution you require for your strategy '''
+        self.SetStartDate(2000, 1, 1)
+        self.SetEndDate(datetime.now().date() - timedelta(1))
+        self.SetCash(25000);
+
+        self.crude = "SCF/CME_CL1_ON"
```
Comment (jaredbroad) [inline Algorithm.Python/QuandlFuturesDataAlgorithm.py]: Please generally add more comments... e.g. "symbol corresponding to the quandl code."

File: Algorithm.Python/QuandlFuturesDataAlgorithm.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        self.Debug(str(self.Time) + str(" Purchased Crude Oil: ") + self.crude)
+
+
+class QuandlFuture(PythonQuandl):
+    '''Custom quandl data type for setting customized value column name. Value column is used for the primary trading calculations and charting.'''
+    def __init__(self):
+        # Define ValueColumnName: cannot be None, Empty or non-existant column name
+        # If ValueColumnName is "Close", do not use PythonQuandl, use Quandl:
+        # self.AddData[QuandlFuture](self.crude, Resolution.Daily)
+        self.ValueColumnName = "Settle"
```
Comment (AlexCatarino) [inline Algorithm.Python/QuandlFuturesDataAlgorithm.py]: Add this comment because it is really important!

**Pull request merged**